.github/workflows/cd-failover-ecs.yml

```
@@ -0,0 +1,95 @@
+name: CD - Failover ECS
+
+on:
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  AWS_REGION: ap-northeast-2
+
+jobs:
+  build-and-push-ecr:
+    name: Build & Push to ECR
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - service: client-api
+            file: ./apps/backend/Dockerfile
+            context: .
+            target: client
+            ecr_repo_name: codedang-client-api
+            build_args: |
+              target=client
+              app_env=production
+
+          - service: admin-api
+            file: ./apps/backend/Dockerfile
+            context: .
+            target: admin
+            ecr_repo_name: codedang-admin-api
+            build_args: |
+              target=admin
+              app_env=production
+
+          - service: iris
+            file: apps/iris/Dockerfile
+            context: apps/iris
+            target: ''
+            ecr_repo_name: codedang-iris
+            build_args: |
+              app_env=stage
+
+          - service: plag
+            file: apps/plag/Dockerfile
+            context: apps/plag
+            target: ''
+            ecr_repo_name: codedang-plag
+            build_args: |
+              app_env=stage
+
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v4
+
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_FOR_ECR_PUSH }}
```

```suggestion
          role-to-assume: ${{ secrets.AWS_ROLE_FOR_ECR_PUSH }} # IAM: GitHubAction-ECRPush
```
실제 어떤 IAM 계정과 연결되는지 주석 추가하면 추후 디버깅에 좋을 것 같습니다